### PR TITLE
Added test for federation package

### PR DESCRIPTION
**Pull request by PascalDR**
- [x] refactory 
- [x] tesunit battery after refactory

### CODE REVIEW

File: pyeudiw/tests/federation/test_static_trust_chain_validator.py [hunk 1]
```diff
@@ -281,3 +286,97 @@
...
+]
+
+def test_is_valid_equals_false():
+    assert StaticTrustChainValidator(
+        invalid_trust_chain, [ta_jwk.serialize()]).is_valid == False
+    
+from unittest.mock import Mock
+    
+def test_retrieve_ec():
+    import pyeudiw.federation.trust_chain_validator as tcv_test
```
Comment (peppelinux) [inline pyeudiw/tests/federation/test_static_trust_chain_validator.py:314]: please move all the imports within the function in the top of the file

the imports must not be within the functions

**Review by peppelinux — CHANGES_REQUESTED** [review]

File: pyeudiw/tests/federation/test_static_trust_chain_validator.py (hunk 1 above)
Comment (PascalDR) [reply pyeudiw/tests/federation/test_static_trust_chain_validator.py]: Resolved in [fix: moved imports](https://github.com/italia/eudi-wallet-it-python/pull/77/commits/293a7f29a4ed235c19103f09cc5e5d8e8e5d221f)

File: pyeudiw/tests/federation/test_static_trust_chain_validator.py [hunk 2]
```diff
@@ -163,6 +163,7 @@
             "homepage_uri": "https://verifier.example.org/home",
             "policy_uri": "https://verifier.example.org/policy",
             "logo_uri": "https://verifier.example.org/static/logo.svg",
+            "federation_fetch_endpoint": "https://verifier.example.org/fetch",
```
Comment (peppelinux) [inline pyeudiw/tests/federation/test_static_trust_chain_validator.py:166]: No, a Leaf cannot have a fetch endpoint

the fetch endpoint MUST be only exposd by a Trust Anchor and its Intermediates

File: pyeudiw/tests/federation/test_static_trust_chain_validator.py [hunk 3]
```diff
@@ -370,4 +371,31 @@ def mock_method(*args, **kwargs):
...
+    ta_signer = JWS(ta_es, alg="RS256", typ="application/entity-statement+jwt")
+    ta_es_signed = ta_signer.sign_compact([ta_jwk])
+        
+    def mock_method_ec(*args, **kwargs):        
+        if args[0] == "https://trust-anchor.example.eu":
+            return [leaf_ec_signed]
+        raise Exception("Wrong issuer")
+    
+    def mock_method_es(*args, **kwargs):        
+        if args[0] == "https://verifier.example.org/fetch":
```
Comment (peppelinux) [inline pyeudiw/tests/federation/test_static_trust_chain_validator.py:394]: no, see my previous example

**Review by peppelinux — CHANGES_REQUESTED** [review]

File: pyeudiw/tests/federation/test_static_trust_chain_validator.py (hunk 2 above)
Comment (PascalDR) [reply pyeudiw/tests/federation/test_static_trust_chain_validator.py]: Resolved in [fix: fixed test test_update_st_es_case_source_endpoint](https://github.com/italia/eudi-wallet-it-python/pull/77/commits/8ce2aaed1759dbce9146541bfdad7493317e8f18)

**Pull request merged**